packages/core-data/src/selectors.js

```diff
@@ -622,5 +622,9 @@ export const hasFetchedAutosaves = createRegistrySelector(
  */
 export const getReferenceByDistinctEdits = createSelector(
 	() => [],
-	( state ) => [ state.undo.length, state.undo.offset ]
+	( state ) => [
+		state.undo.length,
+		state.undo.offset,
+		state.undo.flattenedUndo,
+	]
```

Comment:
Without this change the selector is not being updated while I keep typing in the same paragraph block which means the AutosaveMonitor is unaware of unsaved edits. It seems like a bug to me but I'm curious about your thoughts @youknowriad.

Reply:
Good catch, it's a weird way to detect dirtiness though, I wonder if the component should rely on `__experimentalGetDirtyEntityRecords` instead.

Reply:
It is a weird way. I suppose it was invented to account for the fact that autosaving doesn't make the post "not dirty" which also affects the `isDirty` prop this component receives. I just tested and the same is true for `__experimentalGetDirtyEntityRecords` - after an autosave completes, it still returns a non-empty list.